### DISCUSSION TITLE
How to do end-to-end messagecrypto with Python client

### DISCUSSION THREAD

**Discussion by jiazhai** [apache/pulsar#18757] [Q&A] [answered]
in issue #6834, user report there is no doc for configure and use feature end-to-end messagecrypto in python.
We need to confirm whether this feature is supported in python, and add needed doc.

**Comment by Huanli-Meng** [comment]
@jiazhai , whether end-to-end messagecrypto is supported on Python?

**Comment by devdattakulkarni** [comment]
@jiazhai We are also interested in this. Do you know if this is already supported and just the documentation is missing, or if this feature is yet to be supported in the Python client?

**Comment by tisonkun** [comment]
cc @BewareMyPower @shibd @RobertIndie what's the current status of this feature?

**Comment by BewareMyPower** [comment, marked as answer]
Here is a test that shows how to use e2e encryption: https://github.com/apache/pulsar-client-python/blob/9182dd7ab69266ad0fb630a129a3ef478c1f8ab0/tests/pulsar_test.py#L369

**Comment by tisonkun** [comment]
@BewareMyPower Thank you! I'll convert this thread to a Q&A and select the answer.

Further improvement to add the example to https://pulsar.apache.org/docs/client-libraries-python or elsewhere is possible.
